osquery/tables/networking/linux/tests/namespace_tests.cpp

```diff
@@ -0,0 +1,175 @@
+  if (server_socket == -1) {
+    std::cerr << "Failed to create the socket\n";
+    return 1;
+  }
+
+  if (bind(server_socket,
+           reinterpret_cast<struct sockaddr*>(&server_addr),
+           sizeof(server_addr)) != 0) {
+    std::cerr << "bind() has failed\n";
+    return 1;
+  }
+
+  if (listen(server_socket, 1) != 0) {
+    std::cerr << "listen() has failed\n";
+    return 1;
+  }
+
+  socklen_t client_addr_size = sizeof(client_addr);
+  auto client_socket = accept(server_socket,
+                              reinterpret_cast<struct sockaddr*>(&client_addr),
+                              &client_addr_size);
+  if (client_socket == -1) {
+    std::cerr << "accept() has failed\n";
+    return 1;
+  }
+
+  close(client_socket);
+  close(server_socket);
+
+  return 0;
+}
+} // namespace
+
+namespace osquery {
+namespace tables {
+class NetNamespaceTests : public testing::Test {};
+
+TEST_F(NetNamespaceTests, listening_ports_table) {
+  // Get the current net namespace
+  std::string net_path = "/proc/" + std::to_string(getpid()) + "/ns/net";
+
+  char current_net_namespace[PATH_MAX] = {};
+  if (readlink(net_path.data(), current_net_namespace, PATH_MAX - 1) == -1) {
+    std::cerr << "Failed to acquire the current net namespace. Are namespaces "
+                 "enabled? Skipping this test...";
+    return;
+  }
+
+  // Clone the process while also creating a new net namespace
+  std::uint8_t new_stack[4096U];
+  auto new_stack_ptr = new_stack + sizeof(new_stack) - 1U;
+
+  auto child_id =
+      clone(NetNamespaceTests_CloneEntryPoint, new_stack_ptr, CLONE_NEWNET, 0);
+  EXPECT_NE(child_id, -1);
+  if (child_id == -1) {
+    std::cerr << "clone() has failed; root privileges are required to create a "
+                 "new namespace\n";
+    return;
+  }
+
+  sleep(1);
+
+  // Make sure the child process is running on a different net namespace
+  net_path = "/proc/" + std::to_string(child_id) + "/ns/net";
+
+  char child_net_namespace[PATH_MAX] = {};
+  auto readlink_err =
+      readlink(net_path.data(), child_net_namespace, PATH_MAX - 1);
+  EXPECT_NE(readlink_err, -1);
+  if (readlink_err == -1) {
+    std::cerr
+        << "Failed to acquire the child net namespace (it may have crashed)\n";
+    kill(child_id, SIGKILL);
+    return;
+  }
+
+  // Convert to std::string so that glog can correctly print the contents
+  EXPECT_NE(std::string(current_net_namespace),
+            std::string(child_net_namespace));
+
+  // Make sure that the listening_ports table (which internally queries
+  // process_open_sockets) has picked up our port
+  auto expected_pid = std::to_string(child_id);
+  auto expected_port = std::to_string(kNamespaceTestPort);
+
+  auto results = SQL::selectAllFrom("listening_ports");
+
+  auto kill_err = kill(child_id, SIGKILL);
+  EXPECT_NE(kill_err, -1);
+  if (kill_err == -1) {
+    std::cerr
+        << "The child process could not be killed (it may have crashed)\n";
+    return;
+  }
+
+  // clang-format off
```

Comment:
Why no clang formatting here? Can the lambda not be re-arranged?

Reply:
I can remove it, I thought it was more readable but maybe it's not!